### ISSUE TITLE
Deleting an S3 blob store causes Nexus to attempt to delete the S3 bucket

### ISSUE THREAD

**Issue by jrbeilke** [sonatype/nexus-blobstore-s3#59]
Thanks for creating an issue! Please fill out this form so we can be
sure to have all the information we need, and to minimize back and forth.

* **What are you trying to do?**
Remove unused S3 blob store from Nexus

* **What feature or behavior is this required for?**
Removing S3 blob store requires allowing Nexus to delete S3 bucket

* **How could we solve this issue? (Not knowing is okay!)**
At a minimum this should be added to the documentation and noted in the S3 bucket policy provided. Ideally there would be an option to disable/prevent Nexus from deleting the S3 bucket when an S3 blob store is removed.

* **Anything else?**
Not sure if this repo is still being updated now that the plugin is part of Nexus, but wanted to open an issue here so others were aware.

**Comment by aegershman** [comment]
I am using this feature, and just so I'm aware-- 

You're saying that by deleting the s3 blobstore, it automatically deletes the associated bucket? 

And there's no way to configure it to say "delete the s3-backed blobstore but leave the s3 bucket intact?"

**Comment by jrbeilke** [comment]
Yep, when initializing a new s3 blob store the code checks for an existing bucket, or will create a new bucket if necessary.

On the other hand when removing an s3 blob store the code checks to see if the bucket is empty, removes the nexus metadata, and deletes the bucket.

There is no option to prevent nexus from deleting the s3 bucket, and if permissions do not allow the bucket to be deleted or there are other files in the bucket then nexus fails to remove the blob store. This in turn causes issues like #18 where the blob store page in Nexus is no longer available, and Nexus will fail to start/restart subsequently due to the invalid blob store.
